Gems/PhysX/Code/Mocks/PhysX/MockPhysicsShape.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzFramework/Physics/Shape.h>
+#include <gmock/gmock.h>
+
+namespace UnitTest
+{
+    class MockPhysicsShape : public Physics::Shape
```

Comment:
Seems like there are 2 other places in the engine where a mock of physics shape class is defined: `Gems\ScriptCanvasPhysics\Code\Tests\ScriptCanvasPhysicsTest.cpp` and `Gems\Blast\Code\Tests\Mocks\BlastMocks.h`.

It'd be nice if there is only 1 definition and the others just use it. Since `Shape` is a class of AzFramework, not PhysX, it'd be better that AzFramework tests project provides the mocks and PhysX, Blast and ScriptCanvasPhysics just reference it.

This can be done in another PR.

Reply:
I would have to get an approval for that from sig-core who owns AzFramework section. It would have to be a separate PR but it does sound good. I will file a ticket for it.

Reply:
FYI "AzFramework/Physics" is owned by sig/simulation, not sig/core. But you can totally do it in a subsequent PR.